application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java

```diff
@@ -176,17 +178,29 @@ RestAction<Message> constructChatGptAttempt(ThreadChannel threadChannel,
             return useChatGptFallbackMessage(threadChannel);
         }
 
+        List<String> ids = new CopyOnWriteArrayList<>();
         RestAction<Message> message =
                 mentionGuildSlashCommand(threadChannel.getGuild(), ChatGptCommand.COMMAND_NAME)
                     .map("""
                             Here is an AI assisted attempt to answer your question 🤖. Maybe it helps! \
                             In any case, a human is on the way 👍. To continue talking to the AI, you can use \
                             %s.
                             """::formatted)
-                    .flatMap(threadChannel::sendMessage);
-
-        for (String aiResponse : chatGPTAnswer.get()) {
-            message = message.map(aiResponse::formatted).flatMap(threadChannel::sendMessage);
+                    .flatMap(threadChannel::sendMessage)
+                    .onSuccess(m -> ids.add(m.getId()));
+
+        String[] answers = chatGPTAnswer.get();
+        for (int i = 0; i < answers.length; i++) {
+            if (i == answers.length - 1) {
+                message = message.map(answers[i]::formatted)
+                    .flatMap(m -> threadChannel.sendMessage(m)
+                        .addActionRow(Button.danger(componentIdInteractor
+                            .generateComponentId(ids.toArray(String[]::new)), "Dismiss")));
+                continue;
+            }
```

Comment:
the order of that code section tripped me hard. please put a comment to explain what the block is about